broker/lib/fabrik/DirectorManager.js

```diff
@@ -177,7 +188,205 @@ class DirectorManager extends BaseManager {
     return this.director.getDeploymentIps(deploymentName);
   }
 
+  executePolicy(scheduled, action, deploymentName) {
+    const runOutput = {
+      'shouldRunNow': false
+    };
+    let targetDirectorConfig;
+    return this.director.getDirectorForOperation(action, deploymentName)
+      .then(directorConfig => {
+        targetDirectorConfig = directorConfig;
+        return this.director.getCurrentTasks(action, targetDirectorConfig);
+      })
+      .then(tasksCount => {
+        let currentTasks, maxWorkers;
+        const allTasks = tasksCount.total;
+        const maxTasks = targetDirectorConfig.max_workers;
+        if (allTasks >= maxTasks) {
+          //no slots left anyway
+          runOutput.shouldRunNow = false;
+          return runOutput;
+        }
+        if (scheduled) {
+          currentTasks = tasksCount.scheduled;
+          maxWorkers = targetDirectorConfig.policies.scheduled.max_workers;
+        } else {
+          currentTasks = tasksCount[action];
```

Comment:
potential undefined. use lodash to avoid undefined runtime errors. `_.get(tasksCounts, action, 0)`